### PR TITLE
Add ability to choose a plan in enso devtools

### PR DESCRIPTION
**Pull request by MrFlashAccount**
### Pull Request Description

https://github.com/enso-org/enso/assets/61194245/5655915c-4dee-4e24-90fb-ba026290b4b6

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: app/ide-desktop/lib/dashboard/src/components/Devtools/EnsoDevtools.tsx [hunk 1]
```diff
@@ -37,17 +41,19 @@ const PaywallDevtoolsContext = React.createContext<{
 })
 
 /**
- * Props for the {@link PaywallDevtools} component.
+ * Props for the {@link EnsoDevtools} component.
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/components/Devtools/EnsoDevtools.tsx:44]: avoid using product name in code. consider either `AppDevtools` or `DashboardDevtools` or even simply `Devtools`

File: app/ide-desktop/lib/dashboard/src/components/Devtools/EnsoDevtools.tsx [hunk 2]
```diff
@@ -92,6 +100,56 @@ export function PaywallDevtools(props: PaywallDevtoolsProps) {
...
+            </ariaComponents.Text>
+
+            <ariaComponents.Form
+              gap="small"
+              schema={schema => schema.object({ plan: schema.string() })}
+              defaultValues={{ plan: session.user.plan ?? 'free' }}
+            >
+              {({ form }) => (
+                <>
+                  <ariaComponents.RadioGroup
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/components/Devtools/EnsoDevtools.tsx:114]: i think i would prefer a `ListBox` but i guess this works

File: app/ide-desktop/lib/dashboard/src/providers/AuthProvider.tsx [hunk 3]
```diff
@@ -599,11 +597,10 @@ export default function AuthProvider(props: AuthProviderProps) {
  * @throws {Error} when used outside a {@link AuthProvider}. */
 export function useAuth() {
   const context = React.useContext(AuthContext)
-  if (context == null) {
-    throw new Error('`useAuth` can only be used inside an `<AuthProvider />`.')
-  } else {
-    return context
-  }
+
+  invariant(context != null, 'useAuth must be used within an AuthProvider.')
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/providers/AuthProvider.tsx]: i think maybe we should keep the `` ` ``. idk though

**Review by somebody1234 — APPROVED** [review]
cr passed

File: app/ide-desktop/lib/dashboard/src/components/Devtools/EnsoDevtools.tsx (hunk 2 above)
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/components/Devtools/EnsoDevtools.tsx:114]: i suppose if the listbox is full width then it'd look a little weird

Comment (MrFlashAccount) [reply app/ide-desktop/lib/dashboard/src/components/Devtools/EnsoDevtools.tsx:114]: radio works well when we have less than 5 options (approximately). 
And anyway this is for us, and only for dev mode.

File: app/ide-desktop/lib/dashboard/src/components/Devtools/EnsoDevtools.tsx (hunk 1 above)
Comment (MrFlashAccount) [reply app/ide-desktop/lib/dashboard/src/components/Devtools/EnsoDevtools.tsx:44]: Enso Devtools sounds like a brand, don't you think? In UI it's also "Enso devtools", so I think keeping naming makes sense here. 
CleanShot 2024-07-10 at 16 03 59@2x

**Comment by somebody1234** [comment]
not exactly realted, but is there a reason "return to dashboard" on the upgrade screen is smaller now? specifically, the back arrow is now tiny on my screen.
(also not 100% sure whether it should have a ring around it)

also i think the upgrade screen should have scroll shadows (we have a hook for it, it's like `useOnScroll` or something).
otherwise on screens that are exactly the wrong side then you can't see the enterprise plan (and there's no UI indicating that it exists):
image

- also seems like it's missing a horizontal scrollbar
- oh and also idk about the top margin on the upgrade screen, it makes it look off center imo
- "Upgrade to invite more" is missing a trailing period
- noticed this a while back but forgot to mention it but: most of the paywall features say "Assign users to groups to manage their permissions.", and idk whether they all need to

**Comment by somebody1234** [comment]
is the activity log supposed to be paywalled behind enterprise tier? @PabloBuchu

**Comment by MrFlashAccount** [comment]
@somebody1234 as far as I remember the activity log is available starting from the solo, but with some restrictions. Can't find a related issue though :(

**Comment by somebody1234** [comment]
QA not passed :x::
- if you explicitly toggle a paywall feature, it *permanently* overrides the user plan. (the reset button does not help)
- other than that everything else seems to work fine.

**Comment by MrFlashAccount** [comment]
> if you explicitly toggle a paywall feature, it permanently overrides the user plan. (the reset button does not help)

The Form issue, the state goes to normal if close/open the devtools.

**Comment by somebody1234** [comment]
if it's not feasible to fix it here then i guess QA passes. otherwise perhaps both sections (plan level, and features) should be in mutually exclusive `<details>` elements and expanding one should collapse the other?

**Comment by MrFlashAccount** [comment]
> if it's not feasible to fix it here then i guess QA passes. otherwise perhaps both sections (plan level, and features) should be in mutually exclusive <details> elements and expanding one should collapse the other?

Eventually, we'll add tabs to the dev tools, and at least one tab will contain feature flags (we'll have more, please reach me in DM I'll share with you my ideas) 

So maybe it makes sense to add `<details>` eventually I guess. We'll see

**Comment by MrFlashAccount** [comment]
> not exactly realted, but is there a reason "return to dashboard" on the upgrade screen is smaller now? specifically, the back arrow is now tiny on my screen.

Yeah, because the default size is xsmall... Will fix it as part of https://github.com/orgs/enso-org/projects/13/views/4?pane=issue&itemId=69160465

The same goes for the other issues you've mentioned about the subscribe page

**Comment by MrFlashAccount** [comment]
@somebody1234 shall I merge it?

**Comment by somebody1234** [comment]
sure

**Pull request merged**